cmds/find/find.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/u-root/u-root/pkg/find"
+)
+
+var (
+	perm   = flag.Int("mode", -1, "Permissions")
+	ftype  = flag.String("type", "", "File type")
+	name   = flag.String("name", ".*", "re for name")
+	root   = flag.String("root", "/tmp", "path for root")
+	long   = flag.Bool("l", false, "long listing")
+	ftypes = map[string]os.FileMode{
+		"f": 0,
+		"d": os.ModeDir,
+	}
+)
+
+func main() {
+	flag.Parse()
+	var mask, mode os.FileMode
+	if *perm != -1 {
+		mask = os.ModePerm
+		mode = os.FileMode(*perm)
+	}
+	if *ftype != "" {
+		intType, ok := ftypes[*ftype]
+		if !ok {
+			log.Fatalf("%v is not a valid file type", *ftype)
```

Comment:
Might be nice to show users what the valid types are. Something like this:
```go
 var keys []string
 for key, _ := range ftypes {
   keys = append(keys, key)
 }
 log.Fatalf("%v is not a valid file type\n valid types are %v", *ftype, strings.Join(keys, ","))
```

Reply:
done.